billiard/process.py

```diff
@@ -270,7 +263,7 @@ def _bootstrap(self):
                 except (OSError, ValueError):
                     pass
             old_process = _current_process
-            _set_current_process(self)
+            _current_process = self
```

Comment:
Hmm, I'm pretty sure processName uses current_process by default, Celery 3.1 does have some monkey patching though to make sure this is set correctly on older Python versions.

But it's not just current_process either, there's also register_after_fork, where we want to run the after forkers registered for both multiprocessing and billiard.  I would say we should focus on fixing specific compatibility issues first, unless that becomes unmanageable.